lib/logstash/outputs/opsgenie.rb

```diff
@@ -174,9 +195,32 @@ def executePost(uri, params)
         http.use_ssl = true
         http.verify_mode = OpenSSL::SSL::VERIFY_NONE
       end
-      request = Net::HTTP::Post.new(url.request_uri, initheader = { "Content-Type" =>"application/json", "Authorization" => "GenieKey #{@apiKey}" })
-      request.body = params.to_json
-      response = http.request(request)
+
+      retryCount = 0
+      begin
+        request = Net::HTTP::Post.new(
+          url.request_uri,
+          initheader = { "Content-Type" =>"application/json", "Authorization" => "GenieKey #{@apiKey}" }
+        )
+        request.body = params.to_json
+        response = http.request(request)
+      rescue *HTTP_EXCEPTIONS => e
+        @logger.warn("Silencing exception in logstash opsgenie outputs: [#{e.class}: #{e.message}]")
+        return
+      rescue *RETRYABLE_HTTP_EXCEPTIONS => e
+        if retryCount < HTTP_EXCEPTION_RETRIES
+          retryCount += 1
+          @logger.warn("Caught #{e.class}: #{e.message} - Retrying...")
+          exponentialBackoff(retryCount)
+          retry
+        else
+          @logger.warn(
+            "Silencing exception in logstash opsgenie outputs after #{HTTP_EXCEPTION_RETRIES} retries: [#{e.class}: #{e.message}]"
+          )
+          return
+        end
```

Comment:
We don't want to catch any other exceptions? I.e. a 'catch all' clause? I think it's just a blank `rescue` essentially.

Reply:
hmm I am quite reluctant of catching all exception and silencing them, we may never see what happened since we would just silence them. This is calling http module so catching network errors seems to make sense.